language/bytecode-verifier/src/abstract_state.rs

```diff
@@ -599,96 +415,34 @@ impl AbstractDomain for AbstractState {
-                        BorrowInfo::BorrowedBy(borrowed_by2) => {
-                            borrows.insert(
-                                *nonce,
-                                BorrowInfo::BorrowedBy(
-                                    borrowed_by1.union(borrowed_by2).cloned().collect(),
-                                ),
-                            );
-                        }
-                        BorrowInfo::FieldsBorrowedBy(fields_borrowed_by2) => {
-                            borrows.insert(
-                                *nonce,
-                                BorrowInfo::BorrowedBy(Self::get_union_of_sets(
-                                    borrowed_by1,
-                                    fields_borrowed_by2,
-                                )),
-                            );
-                        }
+                    AbstractType {
+                        signature: self.locals[idx].signature.clone(),
+                        value: AbstractValue::Value(*kind1),
                     },
-                    BorrowInfo::FieldsBorrowedBy(fields_borrowed_by1) => {
-                        match &state.borrows[nonce] {
-                            BorrowInfo::BorrowedBy(borrowed_by2) => {
-                                borrows.insert(
-                                    *nonce,
-                                    BorrowInfo::BorrowedBy(Self::get_union_of_sets(
-                                        borrowed_by2,
-                                        fields_borrowed_by1,
-                                    )),
-                                );
-                            }
-                            BorrowInfo::FieldsBorrowedBy(fields_borrowed_by2) => {
-                                borrows.insert(
-                                    *nonce,
-                                    BorrowInfo::FieldsBorrowedBy(Self::get_union_of_maps(
-                                        fields_borrowed_by1,
-                                        fields_borrowed_by2,
-                                    )),
-                                );
-                            }
-                        }
-                    }
-                }
-            } else {
-                borrows.insert(*nonce, borrow_info.clone());
-            }
-        }
-        for (nonce, borrow_info) in &state.borrows {
-            if !borrows.contains_key(nonce) {
-                borrows.insert(*nonce, borrow_info.clone());
+                );
             }
         }
 
-        let partition = self.partition.join(&state.partition);
-
-        let next_state = AbstractState {
-            locals,
-            globals,
-            borrows,
-            partition,
-        };
-        if next_state == *self {
+        let locals_unchanged = self.locals.keys().all(|idx| locals.contains_key(idx));
+        let borrow_graph_unchanged = self.borrow_graph.subsumes(&state.borrow_graph);
```

Comment:
Simplified Join Proposal: 
change this to `==`

Reply:
Resolved offline.